browser/src/control/Control.SheetsBar.js

```diff
@@ -104,6 +123,86 @@ class SheetsBar {
 		}
 	}
 
+	showSheetListPopup() {
+		const buttonElement = document.getElementById('sheets-buttons-toolbox');
+
+		if (!this.partNames || this.partNames.length === 0) {
+			return;
+		}
+
+		let sheetEntries = [];
+		
+		for (let i = 0; i < this.partNames.length; i++) {
+			if (app.calc.isPartHidden(i)) {
+				continue;
+			}
+
+			let displayText = this.partNames[i];
+			sheetEntries.push({
+				columns: [ { text: displayText } ],
+				row: i,
+				selected: i === this.selectedPart
+			});
+		}
+
+		let json = {
+			id: 'SheetListPopup-dropdown',
+			type: 'dropdown',
+			jsontype: 'dialog',
+			cancellable: true,
+			popupParent: buttonElement,
+			clickToClose: '_POPOVER_',
+			children: [
+				{
+					id: 'container-dropdown',
+					type: 'container',
+					enabled: true,
+					vertical: true,
+					children: [
+						{
+							type: 'treelistbox',
+							id: 'sheetnames',
+							entries: sheetEntries
+						}
+					]
+				}
+			]
+		};
+
+		let popupBuildEvent = {
+			data: json,
+			callback: this.onSelectSheet.bind(this),
+		};
+
+		this.map.fire('jsdialog', popupBuildEvent);
+
+		return json;
+	}
+
+	onSelectSheet(element, action, data, index) {
+		if (element === 'treeview') {
+			if (action === 'select') {
+				// Switch to the selected sheet
+				let partIndex = parseInt(index);
+				if (!isNaN(partIndex) && partIndex !== this.map._docLayer._selectedPart) {
+					this.map._docLayer._clearReferences();
+					this.map.setPart(partIndex, false, true);
+				}
+
+				this.closeSheetListPopup();
+			}
+			return;
+		}
+	}
+
+	closeSheetListPopup() {
+		let popupExists = window.L.DomUtil.get('SheetListPopup-dropdown');
+		if (!popupExists) return;
+
+		this.map.jsdialog.focusToLastElement('SheetListPopup-dropdown');
+		this.map.jsdialog.clearDialog('SheetListPopup-dropdown');
```

Comment:
`clearDialog` shouldn't be used outside of JSDialog.js, let's allow it to clean after itself using native dropdown :)